sklearn/mixture/base.py

```diff
@@ -228,7 +228,8 @@ def fit(self, X, y=None):
             if self.lower_bound_ > max_lower_bound:
                 max_lower_bound = self.lower_bound_
                 best_params = self._get_parameters()
-                best_n_iter = n_iter
+                # Report the number of completed iterations, not 0-indexed
```

Comment:
I would remove the comment

Reply:
Then people will just think this is a mistake.
sklearn code could do with much more inline comments, actually. I also do not like how the current code relies on "leaking" the `n_iter` variable out of the for loop. That is pretty bad code style, but I tried to keep the patch small. And the variable is badly named. The name appears to be the "number of iterations", but it is a 0-based index.
Feel free to do any such changes, add a regression test, etc.

Reply:
Second thought, the comment is ok there. It will remove some ambiguity.

> Then people will just think this is a mistake.
I hope that people try to understand code before to mention this is a mistake :).

> sklearn code could do with much more inline comments, actually
I would argue that code should be self-explanatory. Comments should be there to remove ambiguity.

> And the variable is badly named
True and it brings ambiguity ;)

> I also do not like how the current code relies on "leaking" the n_iter variable out of the for loop. That is pretty bad code style
I don't really see the issue with it.

> Feel free to do any such changes, add a regression test, etc.
We are reviewing code of contributors but do not modify theirs PRs directly (apart of minor changes before merging).